static/js/notifications.js

```diff
@@ -277,6 +277,42 @@ exports.notify_above_composebox = function (note, link_class, link_msg_id, link_
     $('#out-of-view-notification').show();
 };
 
+if (window.electron_bridge !== undefined) {
+    // The code below is for sending a message received from notification reply which
+    // is often refered to as inline reply feature. This is done so desktop app doesn't
+    // have to depend on channel.post for setting crsf_token and narrow.by_topic
+    // to narrow to the message being sent.
+    window.electron_bridge.send_notification_reply_message_supported = true;
+    window.electron_bridge.on_event('send_notification_reply_message', function (message_id, reply) {
+        var message = message_store.get(message_id);
+        var data = {
+            type: message.type,
+            content: reply,
+            to: message.type === 'private' ? message.reply_to : message.stream,
+            topic: util.get_message_topic(message),
+        };
+
+        function success() {
+            narrow.by_topic(message_id, {trigger: 'desktop_inline_reply'});
```

Comment:
Awesome, this is a lot nicer!  But I think this should probably have a conditional on the `message_type`, right?

Reply:
Hmm, I don't follow. Do we only want to narrow if the message is of type `private`? I think it's a good idea to narrow in every case since a reply will be sent.

Reply:
I added a conditional here so we use `narrow.by_topic` if message is of type `stream` and `narrow.by_recipient` otherwise. I didn't catch this during testing because `narrow.by_topic` calls `narrow.by_recipient` internally if the message is not 
 of type `private`.